core/example/03_struct_of_arrays/soa_example.cpp

```diff
@@ -0,0 +1,142 @@
+/****************************************************************************
+ * Copyright (c) 2018 by the Cabana authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cabana_Core.hpp>
+
+#include <iostream>
+
+//---------------------------------------------------------------------------//
+// SoA example.
+//---------------------------------------------------------------------------//
+void soaExample()
+{
+    /*
+      Cabana structs-of-arrays (SoAs) are groups of Tuples with the data
+      specified in an order such that the tuple index is stride-1 in
+      memory. This results in each dimension of each multidimensional tuple
+      member being stored in a contiguous memory block the size of the number
+      of tuples. The number of tuples in an SoA is defined as the "vector
+      length".
+
+      For example, consider declaring the following member type to store two
+      arrays of doubles in each tuple:
+
+          using types = Cabana::MemberTypes<double[3][2],double[4]>;
```

Comment:
It may be more complete and consistent with the code below by adding a scalar member.

Reply:
good call